server/src/main/java/org/apache/druid/metadata/SegmentsMetadataManager.java

```diff
@@ -53,7 +53,20 @@ public interface SegmentsMetadataManager
    */
   int markAsUsedAllNonOvershadowedSegmentsInDataSource(String dataSource);
 
-  int markAsUsedNonOvershadowedSegmentsInInterval(String dataSource, Interval interval);
+  /**
+   * Marks non-overshadowed unused segments for the given interval as used.
+   * @return Number of segments updated
+   */
+  default int markAsUsedNonOvershadowedSegmentsInInterval(String dataSource, Interval interval)
```

Comment:
This default interface method is only used by tests. Should we just remove it in favor of the three argument counterpart? Same for `markAsUsedNonOvershadowedSegmentsInInterval`.